### DISCUSSION TITLE
Ignore blank lines behavior

### DISCUSSION THREAD

**Discussion by JoshClose** [JoshClose/CsvHelper#1637] [General]
I've rewritten the parser and come across something I think should be changed.

When parsing with `IgnoreBlankLines = true`, lines that only contain `\r\n` are skipped. If you set `IgnoreBlankLines = false`, then the parser doesn't skip over those. Currently, a line that has only a `\r\n` will give back an empty array for 0 fields. I think this is actually wrong. Since it's not being ignored it's technically a single field with no data; an empty string.

Does anyone have a compelling reason not to make this change?

**Comment by StefanBertels** [comment]
IMHO an empty non-ignored line is a line with one column with an zero-length string (which might be interpreted as "" or 
null).

This means that there cannot be a line with 0 columns. A line with n columns always has (n-1) column separators (ignoring quoted ones). Vice Versa this means that a line with 0 column separators has 1 columns.
